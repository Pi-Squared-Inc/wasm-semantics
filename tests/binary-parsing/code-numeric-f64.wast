(module $mymodule
  (func (param i32) (result i32)

    i32.const 0
    f64.convert_i32_s
    f64.abs
    drop

    i32.const 0
    f64.convert_i32_s
    f64.neg
    drop

    i32.const 0
    f64.convert_i32_s
    f64.ceil
    drop

    i32.const 0
    f64.convert_i32_s
    f64.floor
    drop

    i32.const 0
    f64.convert_i32_s
    f64.trunc
    drop

    i32.const 0
    f64.convert_i32_s
    f64.nearest
    drop

    i32.const 0
    f64.convert_i32_s
    f64.sqrt
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.add
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.sub
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.mul
    drop

    i32.const 1
    f64.convert_i32_s
    i32.const 1
    f64.convert_i32_s
    f64.div
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.min
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.max
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.copysign
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.eq
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.ne
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.lt
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.gt
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.le
    drop

    i32.const 0
    f64.convert_i32_s
    i32.const 0
    f64.convert_i32_s
    f64.ge
    drop

    (return (i32.const 1))
  )
)
