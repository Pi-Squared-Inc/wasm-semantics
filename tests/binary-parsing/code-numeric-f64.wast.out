<generatedTop>
  <k>
    #module (... types: #type (... type: [ i32  .ValTypes ] -> [ i32  .ValTypes ] , metadata:  )  .EmptyStmts , funcs: #func (... type: 0 , locals: [ .ValTypes ] , body: i32 . const 0  f64 . convert_i32_s  f64 . abs  drop  i32 . const 0  f64 . convert_i32_s  f64 . neg  drop  i32 . const 0  f64 . convert_i32_s  f64 . ceil  drop  i32 . const 0  f64 . convert_i32_s  f64 . floor  drop  i32 . const 0  f64 . convert_i32_s  f64 . trunc  drop  i32 . const 0  f64 . convert_i32_s  f64 . nearest  drop  i32 . const 0  f64 . convert_i32_s  f64 . sqrt  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f32 . add  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f32 . sub  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f32 . mul  drop  i32 . const 1  f64 . convert_i32_s  i32 . const 1  f64 . convert_i32_s  f32 . div  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f32 . min  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f32 . max  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f32 . copysign  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f64 . eq  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f64 . ne  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f64 . lt  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f64 . gt  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f64 . le  drop  i32 . const 0  f64 . convert_i32_s  i32 . const 0  f64 . convert_i32_s  f64 . ge  drop  i32 . const 1  return  .EmptyStmts , metadata: #meta (... id:  , localIds: .Map ) )  .EmptyStmts , tables: .EmptyStmts , mems: .EmptyStmts , globals: .EmptyStmts , elem: .EmptyStmts , data: .EmptyStmts , start: .EmptyStmts , importDefns: .EmptyStmts , exports: .EmptyStmts , metadata: #meta (... id:  , funcIds: .Map , filename: .String ) ) ~> .K
  </k>
  <wasm>
    <instrs>
      .K
    </instrs>
    <valstack>
      .ValStack
    </valstack>
    <curFrame>
      <locals>
        .List
      </locals>
      <curModIdx>
        .Int
      </curModIdx>
    </curFrame>
    <moduleRegistry>
      .Map
    </moduleRegistry>
    <moduleIds>
      .Map
    </moduleIds>
    <moduleInstances>
      .ModuleInstCellMap
    </moduleInstances>
    <nextModuleIdx>
      0
    </nextModuleIdx>
    <mainStore>
      <funcs>
        .FuncDefCellMap
      </funcs>
      <nextFuncAddr>
        0
      </nextFuncAddr>
      <tabs>
        .TabInstCellMap
      </tabs>
      <nextTabAddr>
        0
      </nextTabAddr>
      <mems>
        .List
      </mems>
      <globals>
        .GlobalInstCellMap
      </globals>
      <nextGlobAddr>
        0
      </nextGlobAddr>
      <elems>
        .ElemInstCellMap
      </elems>
      <nextElemAddr>
        0
      </nextElemAddr>
    </mainStore>
    <deterministicMemoryGrowth>
      true
    </deterministicMemoryGrowth>
  </wasm>
</generatedTop>
